bin/release-note-generator.ts

```diff
@@ -0,0 +1,175 @@
+import { exec } from 'node:child_process';
+import { existsSync, writeFile } from 'node:fs';
+import { exit } from 'node:process';
+
+import prompts from 'prompts';
+
+async function run() {
+  const username = await execAsync(
+    // eslint-disable-next-line rulesdir/typography
+    "gh api user --jq '.login'",
+    'To avoid having to enter your username, consider installing the official GitHub CLI (https://github.com/cli/cli) and logging in with `gh auth login`.',
+  );
+  const prNumber =
+    (await getActivePrNumber(username)) ?? (await getNextPrNumber());
+
+  const result = await prompts([
+    {
+      name: 'githubUsername',
+      message: 'Comma-separated GitHub username(s)',
+      type: 'text',
+      initial: username,
+    },
+    {
+      name: 'pullRequestNumber',
+      message: 'PR Number',
+      type: 'number',
+      initial: prNumber,
+    },
+    {
+      name: 'releaseNoteType',
+      message: 'Release Note Type',
+      type: 'select',
+      choices: [
+        { title: 'Features', value: 'Features' },
+        { title: 'Enhancements', value: 'Enhancements' },
+        { title: 'Bugfix', value: 'Bugfix' },
+        { title: 'Maintenance', value: 'Maintenance' },
+      ],
+    },
+    {
+      name: 'oneLineSummary',
+      message: 'Brief Summary',
+      type: 'text',
+    },
+  ]);
+
+  if (
+    !result.githubUsername ||
+    !result.oneLineSummary ||
+    !result.releaseNoteType
+  ) {
+    console.log('All questions must be answered. Exiting');
+    exit(1);
+  }
+
+  const fileContents = getFileContents(
+    result.releaseNoteType,
+    result.githubUsername,
+    result.oneLineSummary,
+  );
+
+  const filepath = `./upcoming-release-notes/${prNumber}.md`;
+  if (existsSync(filepath)) {
+    const { confirm } = await prompts({
+      name: 'confirm',
+      type: 'confirm',
+      message: `This will overwrite the existing release note ${filepath} Are you sure?`,
+    });
+    if (!confirm) {
+      console.log('Exiting');
+      exit(1);
+    }
+  }
+
+  writeFile(filepath, fileContents, err => {
+    if (err) {
+      console.error('Failed to write release note file:', err);
+      exit(1);
+    } else {
+      console.log(
+        `Release note generated successfully: ./upcoming-release-notes/${prNumber}.md`,
+      );
+    }
+  });
+}
+
+// makes an attempt to find an existing open PR from <username>:<branch>
+async function getActivePrNumber(
+  username: string,
+): Promise<number | undefined> {
+  if (!username) {
+    return undefined;
+  }
+  const branchName = await execAsync('git rev-parse --abbrev-ref HEAD');
+  if (!branchName) {
+    return undefined;
+  }
+  const forkHead = `${username}:${branchName}`;
```

Comment:
Noting that this logic doesn't work if you're running it on someone else's PR (source: tried it on this one 😅) but I think that edge case shouldn't normally occur for this script

Reply:
Thought about this. I think the way it's implemented is fine to handle this case though. Like you said, it's an edge case.